pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java

```diff
@@ -403,6 +415,159 @@ private SuccessResponse uploadSegment(@Nullable String tableName, TableType tabl
     }
   }
 
+  // Method used to update a list of segments in batch mode with the METADATA upload type.
+  private SuccessResponse uploadSegments(String tableName, TableType tableType, FormDataMultiPart multiParts,
+      boolean enableParallelPushProtection, boolean allowRefresh, HttpHeaders headers, Request request) {
+    String rawTableName = TableNameBuilder.extractRawTableName(tableName);
+    String tableNameWithType = tableType == TableType.OFFLINE ? TableNameBuilder.OFFLINE.tableNameWithType(rawTableName)
+        : TableNameBuilder.REALTIME.tableNameWithType(rawTableName);
+
+    TableConfig tableConfig = _pinotHelixResourceManager.getTableConfig(tableNameWithType);
```

Comment:
Can we add a log somewhere (either here or in the caller) on the batch size that this API is taking up.